src/xplugd.c

```diff
@@ -33,11 +33,16 @@ char *prognm;
 static char *rcfile(char *arg)
 {
 	glob_t gl;
-	char *rc;
+	char *rc, *config_home;
 	int flags = GLOB_ERR;
 
-	if (!arg)
+	config_home = getenv("XDG_CONFIG_HOME");
+
+	if (!arg && !config_home) {
 		arg = XPLUGRC;
+	} else {
+		arg = strcat(config_home, "/xplugrc");
```

Comment:
This may work for some C library or $ARCH, but we cannot rely on there being enough room for appending text to an environment string returned by `getenv()`.  Also, please use `strlcat()` or `snprintf()` instead of unsafe `strcat()`.